src/sage/rings/polynomial/polynomial_singular_interface.py

```diff
@@ -74,6 +74,9 @@ def _do_singular_init_(singular, base_ring, char, _vars, order):
          //        block   2 : ordering C,
          None)
     """
+    if singular is None:
+        from sage.interfaces.singular import singular
+
     make_ring = lambda s: singular.ring(s, _vars, order=order)
```

Comment:
This does not look right. The caller should provide `singular`.

Reply:
Indeed. I've removed it in ee8875e89cb